packages/swapper/src/swappers/ButterSwap/xhr.ts

```diff
@@ -0,0 +1,214 @@
+import type { Result } from '@sniptt/monads'
+import { Err, Ok } from '@sniptt/monads'
+
+import type { SwapErrorRight } from '../../types'
+import { TradeQuoteError } from '../../types'
+import { makeSwapErrorRight } from '../../utils'
+import type {
+  BridgeInfoApiResponse,
+  BuildTxResponse,
+  FindTokenResponse,
+  RouteAndSwapResponse,
+  RouteResponse,
+  SupportedChainListResponse,
+} from './types'
+import { butterHistoryService } from './utils/butterSwapHistoryService'
+import { butterService } from './utils/butterSwapService'
+
+type ButterSwapPromise<T> = Promise<Result<T, SwapErrorRight>>
+
+/**
+ * @see https://docs.butternetwork.io/butter-swap-integration/butter-api-for-routing/get-supportedchainlist
+ */
+export const getSupportedChainList = async (): ButterSwapPromise<SupportedChainListResponse> => {
+  const result = await butterService.get<SupportedChainListResponse>('/supportedChainList')
+  if (result.isErr()) return Err(result.unwrapErr())
+  // No runtime validation, just return the data
+  return Ok(result.unwrap().data)
+}
+
+/**
+ * @see https://docs.butternetwork.io/butter-swap-integration/butter-api-for-routing/get-findtoken
+ */
+export const findToken = async (
+  chainId: number,
+  address: string,
+): ButterSwapPromise<FindTokenResponse> => {
+  const result = await butterService.get<FindTokenResponse>('/findToken', {
+    params: { chainId, address },
+  })
+  if (result.isErr()) return Err(result.unwrapErr())
+  return Ok(result.unwrap().data)
+}
+
+/**
+ * @see https://docs.butternetwork.io/butter-swap-integration/butter-api-for-routing/get-route
+ */
+export const getRoute = async (
+  fromChainId: number,
+  tokenInAddress: string,
+  toChainId: number,
+  tokenOutAddress: string,
+  amountHumanUnits: string,
+  slippage: string,
+  affiliate: string,
+): ButterSwapPromise<RouteResponse> => {
+  const params: Record<string, any> = {
+    fromChainId,
+    tokenInAddress,
+    toChainId,
+    tokenOutAddress,
+    amount: amountHumanUnits,
+    type: 'exactIn',
+    slippage,
+    entrance: 'Butter+',
+    affiliate,
+  }
+  const result = await butterService.get<RouteResponse>('/route', { params })
+  if (result.isErr()) return Err(result.unwrapErr())
+  const data = result.unwrap().data
+  if (data.errno > 0) {
+    return Err(
+      makeSwapErrorRight({
+        message: `[getRoute] ${data.message}`,
+        code: butterSwapErrorToTradeQuoteError(data.errno),
+      }),
+    )
+  }
+  return Ok(data)
+}
+
+/**
+ * @see https://docs.butternetwork.io/butter-swap-integration/butter-api-for-routing/get-swap
+ */
+export const getBuildTx = async (
```

Comment:
This should be `getQuote` / `getButterQuote`

Reply:
originally discussed with @0xApotheosis and it was like either or, so not gonna change now